web/src/composebox_typeahead.js

```diff
@@ -521,8 +521,16 @@ export function get_person_suggestions(query, opts) {
         if (opts.want_broadcast) {
             persons = [...persons, ...broadcast_mentions()];
         }
-
-        return persons.filter((item) => typeahead_helper.query_matches_person(query, item));
+        // `sort_recipients` and other functions like `user_pill.get_user_ids`
+        // are shared with the pill typeahead which has only users, and we
+        // need a way to differentiate these mentons-or-users from just users,
+        // to help with typing.
+        const person_items = persons.map((person) => ({
+            ...person,
+            type: "user_or_mention",
```

Comment:
This can be a follow-up refactor, but I think the right policy is probably be to eliminate this type value in favor of having this be either a `type: "user"` or a `type: "wildcard_mention"` -- and then types like `UserOrMention` would be a union of the wildcard mention type and the user type. At least, I don' t see a clear reason we'd prefer this way of doing a compound type.